opensrp-core/src/main/java/org/smartregister/util/JsonFormUtils.java

```diff
@@ -353,7 +353,14 @@ private static void createObsFromPopUpValues(Event event, JSONObject jsonObject,
                                     popupJson.put(KEY, secondaryValueKey);
                                     popupJson.put(OPENMRS_ENTITY, CONCEPT);
                                     popupJson.put(OPENMRS_ENTITY_ID, parentOpenMRSAttributes.getString(OPENMRS_ENTITY_ID));
-                                    popupJson.put(VALUE, valueOpenMRSAttribute.getString(OPENMRS_ENTITY_ID));
+                                    if(valueOpenMRSAttribute.has(VALUE))
+                                        popupJson.put(VALUE, valueOpenMRSAttribute.getString(VALUE));
+                                    else
+                                        popupJson.put(VALUE, valueOpenMRSAttribute.getString(OPENMRS_ENTITY_ID));
+                                    if(valueOpenMRSAttribute.has(TEXT))
+                                        popupJson.put(TEXT,valueOpenMRSAttribute.getString(TEXT));
+                                    if(valueOpenMRSAttribute.has(OPTIONS_FIELD_NAME))
+                                        popupJson.put(OPTIONS_FIELD_NAME,valueOpenMRSAttribute.getJSONArray(OPTIONS_FIELD_NAME));
```

Comment:
Please share a sample JSON for the `popupJson`

Reply:
@dubdabasoduba 
{"key":"hiv_test_status","openmrs_entity":"concept","openmrs_entity_id":"165383AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA","value":"done_today","text":"Done today","options":[{"key":"done_today","text":"Done today","type":"done_today","openmrs_entity_parent":"","openmrs_entity":"concept","openmrs_entity_id":"165383AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"}],"type":"extended_radio_button","openmrs_entity_parent":""}

in options only the selected key is added to avoid the extra looping when filling up the human readable values